examples_test/config_test.go

```diff
@@ -33,15 +40,17 @@ func TestConfig_hardwired(t *testing.T) {
 	})
 }
 
-// TestHelloWorld_semiHardWired uses the special "lookup_key" function MuxLookupKey. This function use the configuration
-// option LookupProvidersKey where it expects to find a slice of "lookup_key" functions to use.
+// TestHelloWorld_semiHardWired uses the "lookup_key" function MuxLookupKey. This function use the configuration
+// option LookupKeyFunctions where it expects to find a slice of "lookup_key" functions to use. Those functions form
+// top level hierarchy that is configurable from code. Very useful if you for instance want to introduce different
+// lookup layers such as "global", "environment", and "module" or in other ways build a complex lookup hierarchy that
+// service that goes beyond what can be defined in the yaml configuration.
```

Comment:
aha, so could I use the Mux to create a slice with ConfigLookupKey based resulting functions - one per module and thereby get "data in modules" or is more needed? (I could loop naturally)

Reply:
Almost. You would need a specialized `ConfigLookupKey` function to cater for path resolution of environment and module roots. Hiera doesn't know about those concepts at all right now. We need to invent new options to carry them.